### DISCUSSION TITLE
Attachments no longer uploading (500) after SQLSTATE[HY000] [2006] Error

### DISCUSSION THREAD

**Discussion by DiscoSlug** [firefly-iii/firefly-iii#8864] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

This one's an odd one. I've been using Firefly successfully for months. I haven't required any updates to the background docker or OS for roughly 2 months and have been happily trans actioning my entire life. After 5 successful uses of the importer with CSV files, I received this error error screen (also see attached screenshot):

Could not poll the database: SQLSTATE[HY000] [2006] MySQL server has gone away (Connection: mysql, SQL: select `id`, `name`, `data` from `configuration` where `name` = is demo site and `configuration`.`deleted_at` is null limit 1) This error occured in file /var/www/html/app/Support/Firefly Config.php on line 65 with code 0.
![firefly iii error critical](https://github.com/firefly-iii/firefly-iii/assets/143626901/0dbb0924-2af9-43c2-8a66-631fd0681afe)

Firefly was unresponsive. A reboot appeared to clear it up, as I had multiple containers all go down at the same time. My file paths also disappeared, but returned on reboot. 

I proceeded to login to firefly and see that all was well and working again. After checking one of the files I had uploaded without error messages, this appeared:

![attachment lost firefly](https://github.com/firefly-iii/firefly-iii/assets/143626901/c379c326-6661-4563-884f-b8e351cfd3d9)

So after the first error, firefly was no longer saving attachments. I checked the attachments I had added prior to the error and they are present in the storage folder. Attachments I added after the error, show as if they are present in the attachments section on the transaction page, and link to the url expected, but were never uploaded by firefly. I wasn't able to get logs for adding of those files because I wasn't aware the issue had occurred. After turning debug mode on, the inability to add files solved itself. After turning debug off, the issue has not resumed. 

### Debug information

Debug information generated at 2024-05-04 22:41:20 for Firefly III version **v6.1.15**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.15 / v2.0.14 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.6 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1074, base <span>#</span>83</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>America/New_York + America/New_York</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>info, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 2</td></tr><tr><td>User flags</td><td><span title="Has liabilities">:credit_card:</span> <span title="Has piggy banks">:pig:</span> <span title="Has reconciled">:ledger:</span> <span title="Has rules">:wrench:</span> <span title="Has recurring transactions">:clock130:</span> <span title="Has object groups">:bookmark_tabs:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-01-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-05-04 23:59:59</td></tr><tr><td>View range</td><td>YTD</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36</td></tr></tbody></table>

### Expected behaviour

Attachments showing on transactions page without error message are expected to have been uploaded. 

### Steps to reproduce

Error is reproducible only for the files that have entries but that were never added to the /storage folder.

### Additional info

I can provide the detailed debug logs on request but they contain sensitive information so I would like to not post on the issue.

**Comment by JC5** [comment]
Hey, thanks for opening an issue. 

So, is this still happening or..?

**Comment by DiscoSlug** [comment]
Hi,
It is not currently an ongoing issue, but I wasn't able to find any other instances of it occurring and I haven't got any clue as to what was happening. I was hoping the logs might give insight so that I could prevent reoccurrence.

**Comment by JC5** [comment]
Mmm, indeed. Send me the logs at james@firefly-iii.org. I may not be able to find anything but who knows!

**Comment by DiscoSlug** [comment]
Sent. Thank you!

**Reply by DiscoSlug** [reply, marked as answer]
I seem to have determined the issue. The sata cable to the appdata folder was slightly jiggled out of the port. Just enough to be causing computer wide issues and odd errors, some files were functioning completely normally and some were disappearing. I believe firefly was attempting to upload documents into a section of the drive that was getting lost data packets. SMART showing data packet loss errors equivalent to the number of attachments that I lost. Seems it was a hardware error and not related to firefly.

**Reply by DiscoSlug** [reply]
Also want to say, after deleting the error'ing files from the transactions, the feature that redirects the attachment URL to the transaction it was originally attached to made it a breeze to determine which files went where and correct everything.

**Reply by JC5** [reply]
What a strangely beautiful error, and kudos for finding the issue! A jiggly port is not yet on my list, hah!

Also: glad to be of service in regard to the redirect. I knew it would be helpful one day!
